File: Daily Challenge/Python/Evaluate Parantheses Expression 23.7.2025.py
s = input().strip()
d = t = i = 0
while i < len(s):
    if s[i] == '(': d += 1; i += 1
    elif s[i] == ')': d -= 1; i += 1
    elif s[i].isdigit():
        n = 0
        while i < len(s) and s[i].isdigit():
            n = n * 10 + int(s[i])
            i += 1
        t += n * d
    else: i += 1
print(t)

File: Daily Challenge/Java/Evaluate Parantheses Expression 23.7.2025.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        int i = 0, d = 0, t = 0;
        while (i < s.length()) {
            char ch = s.charAt(i);
            if (ch == '(') {
                d++;
                i++;
            } else if (ch == ')') {
                d--;
                i++;
            } else if (Character.isDigit(ch)) {
                int n = 0;
                while (i < s.length() && Character.isDigit(s.charAt(i))) {
                    n = n * 10 + (s.charAt(i) - '0');
                    i++;
                }
                t += n * d;
            } else {
                i++;
            }
        }
        System.out.println(t);
    }
}